### DISCUSSION TITLE
Extreme amount of light drawcalls on outposts with Nightclub

### DISCUSSION THREAD

**Discussion by cl0ck-byte** [FakeFishGames/Barotrauma#15191] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

![nightclub nightmare](https://github.com/user-attachments/assets/b94dce81-62bf-460b-bba5-78cd6368b239)
On Outposts with Nightclub module the performance goes into the gutter, possibly because some flickering lights cast shadows in the Nightclub. Note that it's the drawcalls and not light updates that cause immense lag.

![nightclub](https://github.com/user-attachments/assets/098b1976-3346-46a2-8982-a00d46d70e36)
That doesn't explain why the performance spikes occur everywhere, even inside the sub. Note that previously described immense drawcall spikes are independent from user's actions - except for when the player is spamming any door for it to close/open repeatedly, which also causes extreme increase in drawcalls.

![noperf](https://github.com/user-attachments/assets/4a81573f-592a-46fd-958f-6d338c67166d)
Disabling shadow-casting lights with PerformanceFix (the issue still occurs on vanilla) seems to resolve the issue.

Limiting light count doesn't fix it, only decreasing shadow resolution helps a little, but it doesn't eliminate the problem.

For whatever reason running the Nightclub module in Sub Editor doesn't cause performance issues, only on Dedicated Server, for which I have no explaination.

Map seed: `NlMF6ync`
Location name: `LocationName.city.33` / `Novaya Moskva Outpost`
Level seed: `LocationName.city.3346`

### Reproduction steps

1. Start multiplayer campaign
2. Dock to an outpost with a nightclub
3. Exist in the outpost location - the issue occurs anywhere on the location (might require loading in the Nightclub module?)

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Multiplayer hosted using a dedicated server

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
> it's the drawcalls and not light updates that cause immense lag

I'm pretty sure it's [CheckConvexHullsInRange](https://github.com/FakeFishGames/Barotrauma/blob/7d5b7a310a52372dac0a8bac90fd2a6a2ab342dc/Barotrauma/BarotraumaClient/ClientSource/Map/Lights/LightSource.cs#L582), which counts as Draw.Map.Lighting

That's why
![Screenshot_4](https://github.com/user-attachments/assets/22d2de1a-247c-4d44-9cfc-2916442cf114)

I've already fixed that [here](https://steamcommunity.com/sharedfiles/filedetails/?id=3365713228)

And Regalis [fixed that in current unstable](https://github.com/FakeFishGames/Barotrauma/discussions/15144#discussioncomment-11263386)

**Comment by Regalis11** [comment]
Duplicate of https://github.com/FakeFishGames/Barotrauma/discussions/15144
